### PR TITLE
stop jobs once all tasks are stopped

### LINKED ISSUE #643: stop jobs where tasks are all stopped
After at least one task within a job has started, once all of the tasks within a job have stopped, the job should be marked as stopped.

### PR DESCRIPTION
**Pull request by bmc-msft**
Fixed #643

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/tasks/main.py [hunk 1]
```diff
@@ -294,22 +281,40 @@ def on_start(self) -> None:
...
-            EventTaskStateUpdated(
-                job_id=self.job_id,
-                task_id=self.task_id,
-                state=self.state,
-                end_time=self.end_time,
-                config=self.config,
+        if self.state == TaskState.stopped:
+            if self.error:
+                send_event(
+                    EventTaskFailed(
```
Comment (chkeita) [inline src/api-service/__app__/onefuzzlib/tasks/main.py:294]: we were sending this event when the state was "stopping" instead of "stopped". Is this change intentional ?

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/tasks/main.py:294]: Yes.  While I was writing this PR, I had shopped a task, got the task stopping event, but files from the task continued to flow in as the agent had stopped the task yet.

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/tasks/main.py]: I pulled the send_state related changes, and I'll make a new PR for that.

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/tasks/main.py]: Note, this additional code is part of #651

**Review by chkeita — APPROVED** [review]
